tempodb/tempodb.go

```diff
@@ -353,6 +356,93 @@ func (rw *readerWriter) Search(ctx context.Context, meta *backend.BlockMeta, req
 	return block.Search(ctx, req, opts)
 }
 
+func (rw *readerWriter) SearchTags(ctx context.Context, meta *backend.BlockMeta, scope string, opts common.SearchOptions) (*tempopb.SearchTagsResponse, error) {
+	attributeScope := traceql.AttributeScopeFromString(scope)
+
+	if attributeScope == traceql.AttributeScopeUnknown {
+		return nil, fmt.Errorf("unknown scope: %s", scope)
+	}
+
+	block, err := encoding.OpenBlock(meta, rw.r)
+	if err != nil {
+		return nil, err
+	}
+	var tags []string
+	tagsMap := make(map[string]bool)
+	rw.cfg.Search.ApplyToOptions(&opts)
+	err = block.SearchTags(ctx, attributeScope, func(t string) {
+		tagsMap[t] = true
+	}, opts)
+
+	for key := range tagsMap {
+		tags = append(tags, key)
+	}
+	return &tempopb.SearchTagsResponse{
+		TagNames: tags,
+	}, err
+}
+
+func (rw *readerWriter) SearchTagValues(ctx context.Context, meta *backend.BlockMeta, tag string, opts common.SearchOptions) ([]string, error) {
+	block, err := encoding.OpenBlock(meta, rw.r)
+	if err != nil {
+		return nil, err
+	}
+	var tags []string
+	tagsMap := make(map[string]bool)
+
+	rw.cfg.Search.ApplyToOptions(&opts)
+	err = block.SearchTagValues(ctx, tag, func(t string) {
+		tagsMap[t] = true
+	}, opts)
+
+	for key := range tagsMap {
+		tags = append(tags, key)
+	}
+	return tags, err
+}
+
+func (rw *readerWriter) SearchTagValuesV2(ctx context.Context, meta *backend.BlockMeta, req *tempopb.SearchTagValuesRequest, opts common.SearchOptions) (*tempopb.SearchTagValuesV2Response, error) {
+	block, err := encoding.OpenBlock(meta, rw.r)
+	if err != nil {
+		return nil, err
+	}
+
+	valueCollector := util.NewDistinctValueCollector[tempopb.TagValue](0, func(v tempopb.TagValue) int { return len(v.Type) + len(v.Value) })
+
+	engine := traceql.NewEngine()
+
+	tag, err := traceql.ParseIdentifier(req.TagName)
+	if err != nil {
+		return nil, err
+	}
+
+	query := traceql.ExtractMatchers(req.Query)
+
+	if !traceql.IsEmptyQuery(query) {
```

Comment:
This is not checking is autocomplete_filtering is enabled or not. I'd add at least a comment mentioning this behaviour.